mne/decoding/time_gen.py

```diff
@@ -227,58 +264,75 @@ def predict(self, epochs):
-                test_times['length'] = self.train_times_['length']
-            if test_times['length'] != self.train_times_['length']:
-                raise ValueError('`train_times` and `test_times` must have '
-                                 'identical `length` keys')
+            test_times['length'] = test_times.get('length',
+                                                  self.train_times_['length'])
             # Make a sliding window for each training time.
             slices_list = list()
-            times_list = list()
-            for t in range(0, len(self.train_times_['slices'])):
+            for __ in range(0, len(self.train_times_['slices'])):
                 test_times_ = _sliding_window(epochs.times, test_times,
                                               epochs.info['sfreq'])
-                times_list += [test_times_['times']]
                 slices_list += [test_times_['slices']]
             test_times = test_times_
             test_times['slices'] = slices_list
-            test_times['times'] = times_list
+        test_times['times'] = [_set_window_time(test, epochs.times)
+                               for test in test_times['slices']]
+
+        for train, tests in zip(self.train_times_['slices'],
+                                test_times['slices']):
+            # The user may define irregular timing. We thus need to ensure
+            # that the dimensionality of each estimator (i.e. training
+            # time) corresponds to the dimensionality of each testing time)
+            if not np.all([len(test) == len(train) for test in tests]):
+                raise ValueError('`train_times` and `test_times` must '
+                                 'have identical lengths')
 
         # Store all testing times parameters
         self.test_times_ = test_times
 
-        # Prepare parallel predictions across time points
+        n_orig_epochs, __, n_times = X.shape
+
+        # Subselects the to-be-predicted epochs so as to manipulate a
+        # contiguous array X by using slices rather than indices.
+        test_epochs = []
+        if self.predict_mode == 'cross-validation':
+            all_test = [ii for train, test in self._cv_splits for ii in test]
+            start = 0
+            for __, test in self._cv_splits:
+                n_test_epochs = len(test)
+                stop = start + n_test_epochs
+                test_epochs.append(slice(start, stop, 1))
+                start += n_test_epochs
+            X = X[all_test]
+
+        # Prepare parallel predictions across testing time points
         # FIXME Note that this means that TimeDecoding.predict isn't parallel
-        parallel, p_time_gen, n_jobs = parallel_func(_predict_slices, n_jobs)
-        n_test_slice = max(len(sl) for sl in self.train_times_['slices'])
+        parallel, p_func, n_jobs = parallel_func(_predict_slices, self.n_jobs)
+        n_test_slice = max(len(sl) for sl in self.test_times_['slices'])
```

Comment:
just `n_slice` ? do you expect the number of slices to be different for train and test? I'm sorry if it's a dumb question. I'm not super familiar with the API